### PR TITLE
Issue #4558: Allow all blocks to have an admin label

### LINKED ISSUE #4558: [UX] Allow *all* blocks to have an admin label
**Description of the need**

If you create a custom block in a layout, but you don't want it to be reusable, there's no way to give that block an admin label that's not displayed on the frontend. Similarly, if you add lots of, say, hero blocks to a layout (all with different visibility conditions), there's no way to differentiate them in the layouts UI if you don't give them a frontend-visible block title.

**Proposed solution**

Just as custom, reusable blocks have admin labels that are seperate from frontend block titles, so should all other blocks have this ability. All blocks should get an optional 'Admin label' field that displays in admin interfaces only (not the frontend).

**Additional information**

This issue is a combination of https://github.com/backdrop/backdrop-issues/issues/4325 (which requests this for existing, pre-made blocks) and https://github.com/backdrop/backdrop-issues/issues/1703 (which requests this for non-reusable, custom blocks). Both have been marked duplicates of this new one.

---
PR by @docwilmot : https://github.com/backdrop/backdrop/pull/3267

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/4558

### CODE REVIEW

File: core/modules/layout/includes/block.class.inc [hunk 1]
```diff
@@ -242,6 +246,22 @@ class Block extends LayoutHandler {
...
       '#maxlength' => 255,
     );
+    $actions_weight = $form['actions']['#weight'];
+    $form['admin_title'] = array(
+      '#type' => 'fieldset',
+      '#title' => t('Administrative title'),
+      '#tree' => FALSE,
+      '#collapsed' => empty($this->settings['admin_title']),
+      '#collapsible' => TRUE,
+      '#weight' => $actions_weight - 1,
```
Comment (klonos) [inline core/modules/layout/includes/block.class.inc:260]: Nice 👍 ...I'm gonna be using this a lot 😄

File: core/modules/layout/includes/block.class.inc [hunk 2]
```diff
@@ -242,6 +246,22 @@ class Block extends LayoutHandler {
...
+    );
+    $form['admin_title']['admin_title'] = array(
+      '#type' => 'textfield',
+      '#default_value' => isset($this->settings['admin_title']) ? $this->settings['admin_title'] : '',
+      '#title' => t('Administrative title'),
+      '#description' => t('Defaults to block title if left blank.'),
```
Comment (klonos) [inline core/modules/layout/includes/block.class.inc:262]: I liked @stpaultim's suggestion:

```suggestion
      '#description' => t('Used to identify the block on the layout. Defaults to the block title if left blank.'),
```

**Comment by docwilmot** [comment]
New sandbox

**Comment by docwilmot** [comment]
New sandbox

**Review by klonos — APPROVED** [review]
LGTM 👍

**Comment by quicksketch** [comment]
Merged into 1.x for 1.17.0.

**Pull request merged**